### DISCUSSION TITLE
Primefaces 13.0.0 not deploying on Payara Glassfish 6 and onwards

### DISCUSSION THREAD

**Discussion by daniloues** [primefaces/community#609] [PrimeFaces] [answered]
Hello, as a preface I'm basically a novice at PrimeFaces and also I've barely used Payara Glassfish, right now I'm developing a Java Web Application on Netbeans and for some reason when I deploy my application on the server with PrimeFaces having a Provided <Scope> it will fail to deploy on Payara Glassfish with the following error:

`java.lang.NoClassDefFoundError: javax/faces/context/FacesContextFactory
parqueo-payara  |       at java.base/java.lang.ClassLoader.defineClass1(Native Method)
parqueo-payara  |       at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
parqueo-payara  |       at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
parqueo-payara  |       at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:524)
parqueo-payara  |       at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:427)
parqueo-payara  |       at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:421)
parqueo-payara  |       at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
parqueo-payara  |       at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:420)
parqueo-payara  |       at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592)
parqueo-payara  |       at com.sun.enterprise.loader.CurrentBeforeParentClassLoader.loadClass(CurrentBeforeParentClassLoader.java:83)
parqueo-payara  |       at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:579)
parqueo-payara  |       at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
parqueo-payara  |       at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1797)
parqueo-payara  |       at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1696)
parqueo-payara  |       at java.base/java.lang.Class.forName0(Native Method)
parqueo-payara  |       at java.base/java.lang.Class.forName(Class.java:467)
parqueo-payara  |       at jakarta.faces.FactoryFinderInstance.getImplGivenPreviousImpl(FactoryFinderInstance.java:410)
parqueo-payara  |       at jakarta.faces.FactoryFinderInstance.getImplementationInstance(FactoryFinderInstance.java:338)
parqueo-payara  |       at jakarta.faces.FactoryFinderInstance.getFactory(FactoryFinderInstance.java:183)
parqueo-payara  |       at jakarta.faces.FactoryFinder.getFactory(FactoryFinder.java:263)
parqueo-payara  |       at com.sun.faces.config.processor.FactoryConfigProcessor.lambda$verifyFactoriesExist$0(FactoryConfigProcessor.java:286)`

This is the logs when trying to run it from a Docker container when I was trying to isolate the issue, I do have the PrimeFaces.jar of the correct version on my Payara6/Glassfish/lib. Also worth mentioning, I changed my deploy from Payara Glassfish to Payara Server and if I build the dependencies first, it will run and deploy, but I'm in need to run it on Docker and I'm not able to do it because it will show me the error always on there.

About the project itself here's the link to my repo: https://github.com/daniloues/primefacesGlassfishApp

**Comment by melloware** [comment]
You must have something else on your classpath as you notice PrimeFaces is no where in that call stack for `java.lang.NoClassDefFoundError: javax/faces/context/FacesContextFactory`

I use PF 13 in an EE10 Jakarta application and it works perfectly.  Check your classpath.

**Reply by daniloues** [reply]
FacesContextFactory seems to be on my EE10 Jakarta dependency but I'm not sure why it fails to locate it when I add the primefaces dependency as provided, even if it's not the primefaces.jar directly, it must be related in some way to why this method would appear missing in the way it does, I thought maybe PrimeFaces was overwriting the method or something alike.

**Comment by daniloues** [comment, marked as answer]
The issue was that PrimeFaces dependency on JSF 2.3.9 was not satisfied by the Payara 6 default configuration, so it had to be set up to use the correct version from the POM.xml.
